### PR TITLE
invoice2data: init at 0.2.93

### PR DESCRIPTION
**Pull request by PsyanticY**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by flokli** [comment]
@Assassinkin thanks for the PR! Can you please rebase your commits, and fix the messages to look like described in CONTRIBUTING.md?

**Comment by Assassinkin** [comment]
@flokli  Thanks. rebased and updated the commit message.

**Comment by flokli** [comment]
According to [invoice2data/input](https://github.com/invoice-x/invoice2data/tree/d9af83d75c02bed90038c21eb5c8aa2f2e0a1288/invoice2data/input), valid input readers are one of `tesseract`, `pdftotext` and `pdfminer` - the first two rely on `tesseract`, `convert` and `pdftotext` being in $PATH, and I'm not sure how `pdfminer` should even work, given the code there tries to import a `pdfminer` python module which is not specified in its dependencies.

I'd suggest to either prefix PATH in the wrapper to contain these two binaries, or patch the src to set the path there explicitly. Maybe it's less hary to go with the first option.

Any idea how `pdfminer` is supposed to work?

**Comment by AmineChikhaoui** [comment]
@GrahamcOfBorg build pythonPackages.invoice2data .

**Comment by Assassinkin** [comment]
@flokli Thanks for the review. I Included the binaries in the PATH as you suggested. For `pdfminer`, I couldn't really find how it is being imported there, I opened an issue in the project repo.

**Comment by flokli** [comment]
upstream issue for the pdfminer input reader: https://github.com/invoice-x/invoice2data/issues/132

File: pkgs/development/python-modules/invoice2data/default.nix
Comment (flokli) [inline pkgs/development/python-modules/invoice2data/default.nix]: This doesn't work, you need to just add the path to packages 'bin' directory, not the executable itself.
For multiple packages, this can also look like this:

```nix
makeWrapperArgs = ["--prefix" "PATH" ":" "${stdenv.lib.makeBinPath [ imagemagick xpdf ]}"];
```

File: pkgs/development/python-modules/invoice2data/default.nix
Comment (flokli) [inline pkgs/development/python-modules/invoice2data/default.nix]: I missed the meta block in my last review. Can you squash that into the first commit? (or squash everything together)

**Review by flokli — CHANGES_REQUESTED** [review]

**Comment by Assassinkin** [comment]
@flokli  Rebased the branch. 
The module is using `pdfminer.six` instead of `pdfminer` which is available only for python27. And the package owner is commenting it ( `pdfminer.six`) in the requirements file.
`pdfminer.six`  is not currently available in nixpkgs though.

**Comment by Assassinkin** [comment]
One of the maintainer of the package replyed on  invoice-x/invoice2data#132 : 
> It doesn't work very well and is only optional. So it's not in the requirements.

I think we can either put this PR on hold till it get fixed or we can proceed without it as it is optional and will require `pdfminer.six` to be packaged.
cc @flokli

**Comment by flokli** [comment]
I'd say it's fine to leave it broken, maybe we should add a comment in our .nix file, in case sb wonders :-)

Can you test and bump to `v0.2.93`? This contains a fix for a https://github.com/invoice-x/invoice2data/pull/134 I found out about yesterday, while trying to test it.

**Comment by Assassinkin** [comment]
@FlashKorten ^^

**Comment by xeji** [comment]
@GrahamcOfBorg build pythonPackages.invoice2data

File: pkgs/development/python-modules/invoice2data/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/invoice2data/default.nix]: Patching the code is the preferred solution, but this is "ok" as well :)

File: pkgs/development/python-modules/invoice2data/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/invoice2data/default.nix]: Remove ` - invoice2data`

**Review by FRidh — CHANGES_REQUESTED** [review]
Is this primarily used as application or library? I saw the README showing a lot of cli examples, and one brief mention of using it as a library.

**Comment by Assassinkin** [comment]
@FRidh Not really sure if it is used as a lib or an application, but according to the [use case of the package](http://193.40.244.77/idu0080/wp-content/uploads/2016/03/UBL%20ODOO%20connect%202016.pdf) i would say it is more used as an app. 
Thanks for the review.

**Comment by FRidh** [comment]
@Assassinkin applications are kept out of `python-packages.nix` and elsewhere in the Nixpkgs tree.

**Comment by Assassinkin** [comment]
@FRidh You mean it should be under another "section" or shouldn't be in `Nixpkgs` at all ?

**Comment by FRidh** [comment]
@Assassinkin right, another section.

**Comment by Assassinkin** [comment]
@FRidh Thanks. I think  `pkgs/tools/text` is the most suitable place in the tree ?

**Comment by FRidh** [comment]
@GrahamcOfBorg build invoice2data

**Comment by Assassinkin** [comment]
@GrahamcOfBorg build invoice2data

**Comment by FRidh** [comment]
@GrahamcOfBorg build invoice2data

**Comment by AmineChikhaoui** [comment]
@GrahamcOfBorg build invoice2data

**Comment by flokli** [comment]
failures on darwin and aarch64 seem to be unrelated.

@Assassinkin: given #41207 was merged, do you want to add it to this package as well?

**Comment by Assassinkin** [comment]
Sorry about that closing and restarting with a new branch

**Pull request closed without merging**
